browser.js

```diff
@@ -26,13 +32,47 @@ function randomBytes (size, cb) {
   }
 
   // XXX: phantomjs doesn't like a buffer being passed here
-  var bytes = Buffer.from(rawBytes.buffer)
+  return Buffer.from(rawBytes.buffer)
+}
+
+function randomBytes (size, cb) {
+  if (isNaN(size) || size < 0) {
+    throw new Error('size must be a number >= 0')
+  }
+
+  if (size > MAX_UINT32) {
+    throw new Error('Requested too many random bytes - the maximum allowed are ' + MAX_UINT32)
+  }
+
+  var bytes
+
+  if (size > MAX_BYTES) {
+    // we can only generate MAX_BYTES at a time
+    bytes = Buffer.alloc(size)
+
+    for (var read = 0; read < size;) {
+      var next = size - read
+
+      if (next > MAX_BYTES) {
+        next = MAX_BYTES
+      }
+
+      generateBytes(next).copy(bytes, read)
```

Comment:
this is still creating multiple buffers and copying them around, if you create 1 buffer and then use the slice method it creates a view of the parent buffer without actually allocating a new one and changes to that view are reflected in the parent